### DISCUSSION TITLE
can we get the query built in an array format ?

### DISCUSSION THREAD

**Discussion by Gabriel4626** [Jeroen-G/Explorer#185] [Q&A]
can we get the query built in an array format, to use it as a search parameter of the elasticsearch/elasticsearch lib?

**Comment by Jeroen-G** [comment]
What would you like to have changed exactly? I don't think I understand your idea. 🤔

**Reply by Gabriel4626** [reply]
I like your lib and have been using it since the beginning of a project. But I got to a point where I have to get the documents directly from elasticsearch because there are too many joins. So I used the elasticsearch/elasticsearch dependency and it doesn't have a query builder. I tried to use the Explorer one. And for the moment, I use it this way, as I'm in a bit of a hurry:


        $host = env('EXPLORER_ES_HOST', 'localhost');

        $port = env('EXPLORER_ES_PORT', '9200');

        $scheme = env('EXPLORER_ES_SHEME', 'http');

        $this->client = ClientBuilder::create()
            ->setHosts(['' . $scheme . '://' . $host . ':' . $port . ''])
            ->build();


...
..

        $parentBoolQuery = new BoolQuery();

        $boolQuery = new BoolQuery();

        $parentBoolQuery->must(new Terms('fk_mill', array_map(function ($v) {
            return (string) $v;
        }, $mills)));
...
...

        $params = [
            'index' => 'datasheet',
            'body'  => [
                "size" => $limit,
                "from" => $page == 1 ? 0 : $limit * ($page - 1),
                'query' => [
                    'bool' => [
                        'must' => [
                            [
                                "query_string" => [
                                    "query" => "*"
                                ]
                            ]
                        ],
                        'should' => [],
                        'filter' => [],
                    ]
                ]
            ]
        ];

// here I get the query built with BoolQuery and add it in paramatre

$query = $parentBoolQuery->build();

        foreach ($query['bool'] as $key => $param) {
            array_push($params['body']['query']['bool'][$key], ...$param);
        }

$this->client->search($param)



However, I would like to be able to do all this only with Explorer, or at least build the complete query and use it as a search parameter with elasticsearch/elasticsearch

**Comment by Jeroen-G** [comment]
You can create the queries with the Syntax classes from the package and then give it to the [adapter](https://github.com/Jeroen-G/Explorer/blob/master/src/Infrastructure/Elastic/ElasticDocumentAdapter.php) to search ES.

Basically that is what this package does to connect it to Scout: https://github.com/Jeroen-G/Explorer/blob/master/src/Infrastructure/Elastic/ElasticDocumentAdapter.php

**Reply by Gabriel4626** [reply]
thanks this is what i needed

**Reply by Gabriel4626** [reply]
thanks this is what i needed
